oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/config/annotation/web/configurers/OAuth2AuthorizationServerConfigurer.java

```diff
@@ -359,8 +403,6 @@ private Map<Class<? extends AbstractOAuth2Configurer>, AbstractOAuth2Configurer>
 		configurers.put(OAuth2TokenEndpointConfigurer.class, new OAuth2TokenEndpointConfigurer(this::postProcess));
 		configurers.put(OAuth2TokenIntrospectionEndpointConfigurer.class, new OAuth2TokenIntrospectionEndpointConfigurer(this::postProcess));
 		configurers.put(OAuth2TokenRevocationEndpointConfigurer.class, new OAuth2TokenRevocationEndpointConfigurer(this::postProcess));
-		configurers.put(OAuth2DeviceAuthorizationEndpointConfigurer.class, new OAuth2DeviceAuthorizationEndpointConfigurer(this::postProcess));
```

Comment:
This change breaks (backward) compatibility since it's not disabled by default. Instead of disabling by default, there needs to be a configurable way of disabling the feature.

Reply:
Make sense.

Reply:
Do we need to provider a configuration(e.g. disableDeviceAuthorizationFlow) and default value is true? Any setting we can refer to?

Reply:
@jgrandja ，
Adding a configuration `deviceAuthorizationEndpoint.enableDeviceAuthorizationEndpoint(true);` for  enable/disable device authorization flow. The default value of `deviceAuthorizationEndpoint.enableDeviceAuthorizationEndpoint ` is true which enable by default for backward compatibility.